packages/expo-notifications/android/src/main/java/expo/modules/notifications/service/delegates/SharedPreferencesNotificationCategoriesStore.kt

```diff
@@ -0,0 +1,156 @@
+package expo.modules.notifications.service.delegates
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Base64
+import expo.modules.notifications.notifications.model.NotificationCategory
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.io.InvalidClassException
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+
+/**
+ * A fairly straightforward [SharedPreferences] wrapper to be used by [NotificationSchedulingHelper].
+ * Saves and reads notification category information (identifiers, actions, and options) to and from persistent storage.
+ *
+ *
+ * A notification category with identifier = 123abc will be persisted under key:
+ * [SharedPreferencesNotificationCategoriesStore.NOTIFICATION_CATEGORY_KEY_PREFIX]123abc
+ */
+class SharedPreferencesNotificationCategoriesStore(context: Context) {
+  companion object {
+    private const val SHARED_PREFERENCES_NAME = "expo.modules.notifications.SharedPreferencesNotificationCategoriesStore"
+    private const val NOTIFICATION_CATEGORY_KEY_PREFIX = "notification_category-"
+  }
+
+  private val sharedPreferences = context.getSharedPreferences(SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE)
+
+  /**
+   * Fetches notification category info for given identifier.
+   *
+   * @param identifier Identifier of the category.
+   * @return Category information: actions and options.
+   * @throws JSONException          Thrown if notification category could not be interpreted as a JSON object.
+   * @throws IOException            Thrown if there is an error when fetching the category from storage.
+   * @throws ClassNotFoundException Thrown if there is an error when interpreting the category fetched from storage.
+   */
+  @Throws(IOException::class, ClassNotFoundException::class)
+  fun getNotificationCategory(identifier: String) =
+    deserializeNotificationCategory(
+      sharedPreferences.getString(
+        preferencesNotificationCategoryKey(identifier),
+        null
+      )
+    )
+
+  /**
+   * Fetches all categories, ignoring invalid ones.
+   *
+   *
+   * Goes through all the [SharedPreferences] entries, interpreting only the ones conforming
+   * to the expected format.
+   *
+   * @return Map with identifiers as keys and category info as values
+   */
+  val allNotificationCategories: Collection<NotificationCategory>
+    get() =
+      sharedPreferences.all.mapNotNull { (key, value) ->
+        if (key.startsWith(NOTIFICATION_CATEGORY_KEY_PREFIX)) {
+          try {
+            return@mapNotNull deserializeNotificationCategory(value as String?)
+          } catch (e: ClassNotFoundException) {
+            // do nothing
+          } catch (e: IOException) {
+            // do nothing
+          }
+        }
+        null
+      }
```

Comment:
I believe that's the perfect moment for adding this one: https://developer.android.com/kotlin/ktx

Reply:
And what would you say for writing it this way:
```suggestion
      sharedPreferences.all
        .filter(key -> key.startsWith(NOTIFICATION_CATEGORY_KEY_PREFIX))
        .mapNotNull { (key, value) -> 
          return try {
            deserializeNotificationCategory(value as String?)
          } catch (e: ClassNotFoundException) {
            null
          } catch (e: IOException) {
            null
          }
        }
```
less indents & `if`s, better readability 😉